### ISSUE TITLE
MD024 (Same header content) Triggers even if (manually set) anchors differ

### ISSUE THREAD

**Issue by hoijui** [markdownlint/markdownlint#378]
## Description

If headers have the same content within one document, the auto generated anchors for them would be the same, so there is no way to link specifically to one of these headers. To fix this, we have MD024.
In some markdown flavors however, it is possible to manually set anchors, this way:

```
# Some text {#some-text-1}

## Some text {#some-text-2}
```

This solves the potential problem that MD024 highlights, but the rule still triggers.
In my eyes, this is a false positive, and it should not trigger.

### Environment

**MDL Version**
0.11.0

### Expected Behavior

MD024 should not trigger if the header contents are same, but the anchors differ;
e.g one or both of them are manually set, and as a result the anchors differ.

### Actual Behavior

... MD024 *does* trigger if anchors differ, but header content is otherwise equal.

## Replication Case

```
echo "# Some text {#some-text-1}

                   ## Some text {#some-text-2}" | mdl
```
->

```
stdin):3: MD024 Multiple headers with the same content
```
